### ISSUE TITLE
SwaggerParser.cs is getting NullRef exception with the latest MariaDB spec

### ISSUE THREAD

**Issue by yantang-msft** [projectkudu/AzureResourceExplorer#256]
We copied the Swagger Parser code to our project long time ago and recently noticed that it's not working with the latest MariaDB swagger spec. After investigation, we found the reason is the ```references``` object at [here](https://github.com/projectkudu/AzureResourceExplorer/blob/master/SwaggerParser/SwaggerParser.cs#L64) got modified during iteration.

Here is why the references object got modified. Basically, MariaDB.json defined "TrackedResource" in the [spec](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json#L1310). At the same time, MariaDB.json references [common-types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L45) where the TrackedResource is also defined. And during the expansion of the swagger spec, the "TrackedResource" defined in MariaDB.json is override by the "TrackedResource" defined in common-types because of this piece of logic here: https://github.com/projectkudu/AzureResourceExplorer/blob/master/SwaggerParser/SwaggerParser.cs#L114

I don't know why the logic is here in the first place, and I think it can be removed completely.
The reason is the intention of ```EnsureCompleteDefinitionIsPresent``` is to pull definitions from external files. i.e., if File1.Def1 references File2.Def2, it will copy Def2 to File1. But [this piece of logic](https://github.com/projectkudu/AzureResourceExplorer/blob/master/SwaggerParser/SwaggerParser.cs#L114) is saying if File2.Def3 references File2.Def2, then copy Def3 to File1. But since File1 is not referencing Def3, why bother copy Def3 to File1? And for MariaDB, since Def1 has the same name as Def3 (TrackedResource) and thus results in the issue we saw.

We have couple ways to workaround this issue, e.g., update MariaDB.json to reference TrackedResource from common-types. Or do a ToList() on the ```references``` enumerable. But still I think removing this logic (https://github.com/projectkudu/AzureResourceExplorer/blob/master/SwaggerParser/SwaggerParser.cs#L114) is the right fix. It would be great if we can get a confirmation from the original author.

**Comment by balag0** [comment]
Is the definition for TrackedResource duplicated or is there some difference between the definitions in common-types and the RP's json? Is the new swagger available at https://github.com/Azure/azure-rest-api-specs?

**Comment by yantang-msft** [comment]
@balag0 Yes, you can try with the latest swagger at  https://github.com/Azure/azure-rest-api-specs, I pasted the exact link of the two files that is causing the problem.
The TrackedResource is duplicated, so essentially the RP's json can reference TrackedResource from common-types to get around this issue. But I think it reveals an issue in the swagger parser, or the logic is by design that I didn't get.
